pyvista/core/grid.py

```diff
@@ -75,6 +75,22 @@ def _get_attrs(self):
         attrs.append(('Dimensions', self.dimensions, '{:d}, {:d}, {:d}'))
         return attrs
 
+    def is_0d(self):
+        dims = np.asarray(self.dimensions)
+        return (dims == 1).all()
+
+    def is_1d(self):
+        dims = np.asarray(self.dimensions)
+        return (dims > 1).sum() == 1
+
+    def is_2d(self):
+        dims = np.asarray(self.dimensions)
+        return (dims > 1).sum() == 2
+
+    def is_3d(self):
+        dims = np.asarray(self.dimensions)
+        return (dims > 1).all()
+
```

Comment:
We could probably make these properties instead of methods. And maybe combine them into one instead of separate? Also, to speed things up and avoid the need to add formal docstrings and examples, for now you can also make these private for the sake of getting it implemented with the filters, and possibly make it a public API in a follow-up PR. E.g.
```suggestion
    @property
    def _dimensionality(self):
        dims = np.asarray(self.dimensions)
        return float(3 - (dims == 1).sum())
        
```

Reply:
If it's not needed we should remove it. 

Using property vs. method can be subjective. Since this method is not expensive to compute and is unlikely to require parameterization in the future, I think it's a good candidate for `@property`.

As a side not I find `is_integer()` to be poorly named. A float is not an integer! `is_whole()` or  `is_intlike()` may have been better. And I would have probably made this a property too. But alas I am not a python dev, maybe this was implemented like this for other reasons I have not considered.